k8s-deploy/pkg/cli/support_bundle.go

```diff
@@ -56,228 +50,132 @@ func SupportBundleCollectCommand() *cli.Command {
 				Value:   200,
 				Usage:   "Specify how many rows to record.",
 			},
+			&cli.StringFlag{
+				Name:    "kubeconfig",
+				Aliases: []string{"c"},
+				Value:   "",
+				Usage:   "Specify the kubeconfig.",
+			},
+			&cli.StringFlag{
+				Name:  "collectDir",
+				Value: "./kubefate-supportbundle",
+				Usage: "Specify where to save temporary data.",
+			},
+			&cli.StringFlag{
+				Name:  "packDir",
+				Value: "./",
+				Usage: "Specify where to pack zip file.",
+			},
 		},
 		Usage: "Collect data",
 		Action: func(c *cli.Context) error {
 			tail := c.Int("tail")
-			return Collect(tail)
+			kubeconfig := c.String("kubeconfig")
+			if kubeconfig == "" {
+				kubeconfig = client.GetKubeconfig()
+			}
+			collectDir := c.String("collectDir")
+			packDir := c.String("packDir")
+			if err := initBundler(kubeconfig, collectDir, packDir); err != nil {
+				return err
+			}
+			namespaceList, err := bundler.Client.GetNamespaceList()
+			if err != nil {
+				return err
+			}
+			var namespaces []string
+			namespacePrompt := &survey.MultiSelect{
+				Message: "Choose namespaces to troubleShooting:",
+				Options: client.NamespaceListToNames(namespaceList),
+				Default: []string{"kube-fate"},
+			}
+			survey.AskOne(namespacePrompt, &namespaces)
+
+			for _, namespace := range namespaces {
+				err := CollectInNamespace(bundler, namespace, tail)
+				if err != nil {
+					fmt.Println(err)
+				}
+			}
+			fmt.Printf("You can attach additional files about this issue into %s\n", collectDir)
+			return nil
 		},
 	}
 }
 
 func SupportBundlePackCommand() *cli.Command {
 	return &cli.Command{
-		Name:  "pack",
-		Flags: []cli.Flag{},
+		Name: "pack",
+		Flags: []cli.Flag{
+			&cli.IntFlag{
+				Name:    "tail",
+				Aliases: []string{"t"},
+				Value:   200,
+				Usage:   "Specify how many rows to record.",
+			},
+			&cli.StringFlag{
+				Name:    "kubeconfig",
+				Aliases: []string{"c"},
+				Value:   "",
+				Usage:   "Specify the kubeconfig.",
+			},
+			&cli.StringFlag{
+				Name:  "collectDir",
+				Value: "./kubefate-supportbundle",
+				Usage: "Specify where to save temporary data.",
+			},
+			&cli.StringFlag{
+				Name:  "packDir",
+				Value: "./",
+				Usage: "Specify where to pack zip file.",
+			},
+		},
 		Usage: "Pack data",
 		Action: func(c *cli.Context) error {
-			return Pack()
+			kubeconfig := c.String("kubeconfig")
+			if kubeconfig == "" {
+				kubeconfig = client.GetKubeconfig()
+			}
+			collectDir := c.String("collectDir")
+			packDir := c.String("packDir")
+			if err := initBundler(kubeconfig, collectDir, packDir); err != nil {
```

Comment:
The shared variable `bundle` does not appear to survive between collect and pack. So I have to initialize twice.